tests/unit/models/gpflow/test_models.py

```diff
@@ -1618,6 +1619,57 @@ def test_sparse_variational_pairwise_covariance_for_non_whitened(
     np.testing.assert_allclose(expected_covariance, actual_covariance, atol=1e-4)
 
 
+class DummyInducingPointSelector(InducingPointSelector[GPflowPredictor]):
+    def __init__(self, new_inducing_points: TensorType, recalc_every_model_update: bool = True):
+        super().__init__(recalc_every_model_update)
+        self._new_inducing_points = new_inducing_points
+
+    def _recalculate_inducing_points(
+        self, M: int, model: ProbabilisticModelType, dataset: Dataset
+    ) -> TensorType:
+        return self._new_inducing_points
+
+
+@random_seed
+@pytest.mark.parametrize("whiten", [False, True])
+def test_sparse_variational_inducing_updates_preserves_posterior(
+    whiten: bool,
+) -> None:
+    x = tf.constant(np.linspace(0.0, 1.0, 8).reshape(-1, 1), dtype=gpflow.default_float())
+    y1 = fnc_3x_plus_10(x)
+
+    num_inducing_points = 4
+    xnew = tf.constant(
+        np.linspace(0.31, 0.77, num_inducing_points).reshape(-1, 1), dtype=gpflow.default_float()
+    )
+
+    svgp = svgp_model_with_mean(x, y1, whiten, num_inducing_points)
+    inducing_point_selector = DummyInducingPointSelector(xnew)
+    model = SparseVariational(
+        svgp,
+        BatchOptimizer(tf.optimizers.Adam(), max_iter=3, batch_size=10),
+        inducing_point_selector=inducing_point_selector,
+    )
+
+    model.optimize(Dataset(x, y1))
```

Comment:
I wonder if it might be sensible to disable training of the `Linear` mean function, as we don't care about the quality of the fit, but we do need some mean function contribution to check that this PR is fixing something.

Reply:
I manually checked that the Linear contribution doesn't go to zero and the tests would fail without the fix

Reply:
OK changed my mind so this is deterministic now